src/plugins/smooth-scroll-mixin.js

```diff
@@ -0,0 +1,36 @@
+export default {
+	methods: {
+		smoothScrollTo({ yPosition, millisecondsToAnimate = 250, stops }) {
+			const startingPosition = window.scrollY;
+			const endTime = (new Date()).getTime() + millisecondsToAnimate;
+			const numberOfStops = stops || Math.ceil(millisecondsToAnimate / 2);
+			const yDifference = startingPosition - yPosition;
+
+			let animationComplete = false;
+
+			for (let i = 0; i < numberOfStops; i += 1) {
+				// eslint-disable-next-line no-loop-func
+				setTimeout(() => {
```

Comment:
I think we should probably convert this setTimeout to use requestAnimationFrame instead. You may also want to create an instance of either setTimeout/requestAnimationFrame that you can stop/cancel/clear out when done.

Some links:
https://flaviocopes.com/requestanimationframe/
https://stackoverflow.com/questions/38709923/why-is-requestanimationframe-better-than-setinterval-or-settimeout

Reply:
Very cool, I'll do some research